Sources/UIKit/PortalCarouselView.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  PortalCarouselView.swift
+//  PortalView
+//
+//  Created by Cristian Ames on 4/17/17.
+//  Copyright © 2017 Guido Marucci Blas. All rights reserved.
+//
+import UIKit
+
+public final class PortalCarouselView<MessageType, CustomComponentRendererType: UIKitCustomComponentRenderer>: PortalCollectionView<MessageType, CustomComponentRendererType>
+    where CustomComponentRendererType.MessageType == MessageType {
+    
+    public var isSnapToCellEnabled: Bool = false
+    
+    fileprivate let onSelectionChange: (ZipListShiftOperation) -> MessageType?
+    fileprivate var lastOffset: CGFloat = 0
+    fileprivate var selectedIndex: Int = 0
+    
+    public override init(items: [CollectionItemProperties<MessageType>], customComponentRenderer: CustomComponentRendererType, layoutEngine: LayoutEngine, layout: UICollectionViewLayout) {
+        onSelectionChange = { _ in .none }
+        super.init(items: items, customComponentRenderer: customComponentRenderer, layoutEngine: layoutEngine, layout: layout)
+    }
+    
+    public init(items: ZipList<CarouselItemProperties<MessageType>>?, customComponentRenderer: CustomComponentRendererType, layoutEngine: LayoutEngine, layout: UICollectionViewLayout, onSelectionChange: @escaping (ZipListShiftOperation) -> MessageType?) {
+        if let items = items {
+            let transform = { (item: CarouselItemProperties) -> CollectionItemProperties<MessageType> in
+                return collectionItem(
+                    onTap: item.onTap,
+                    identifier: item.identifier,
+                    renderer: item.renderer)
+            }
+            selectedIndex = Int(items.centerIndex)
+            self.onSelectionChange = onSelectionChange
+            super.init(items: items.map(transform), customComponentRenderer: customComponentRenderer, layoutEngine: layoutEngine, layout: layout)
+            scrollToItem(self.selectedIndex, animated: false)
+        } else {
+            self.onSelectionChange = onSelectionChange
+            super.init(items: [], customComponentRenderer: customComponentRenderer, layoutEngine: layoutEngine, layout: layout)
+        }
+        
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    public func scrollViewWillBeginDragging(_ scrollView: UIScrollView) {
+        lastOffset = scrollView.contentOffset.x
+    }
+    
+    public func scrollViewWillEndDragging(_ scrollView: UIScrollView, withVelocity velocity: CGPoint,
+                                          targetContentOffset: UnsafeMutablePointer<CGPoint>) {
+        guard isSnapToCellEnabled else { return }
```

Comment:
The problem is that because of the way we are handling now the scroll to detect the index, That is not possible. After scrolling we add or remove one to the counter of selected. We are not detecting which element is in screen at this moment and this is something we should do to provide this feature while `snapToCell` is disabled. I recommend we add this feature in another branch, maybe with the support of a vertical carousel.